### PR TITLE
Daemonset support

### PR DESCRIPTION
**Pull request by PiotrProkop**
This change is [Reviewable](https://reviewable.io/reviews/mirantis/k8s-appcontroller/67)

### CODE REVIEW

**Comment by PiotrProkop** [comment]
Work in progress.

File: scheduler/scheduler.go [hunk 1]
```diff
@@ -120,6 +120,17 @@ func newResourceForReplicaSet(name string, resDefs []client.ResourceDefinition,
...
 
+func newResourceForDaemonSet(name string, resDefs []client.ResourceDefinition, c client.Interface) resources.Resource {
+	for _, rd := range resDefs {
+		if rd.DaemonSet != nil && rd.DaemonSet.Name == name {
+			log.Println("Found resource definition for daemon set", name)
+			return resources.NewDaemonSet(rd.DaemonSet, c.DaemonSets())
+		}
+	}
+
+	log.Printf("Resource definition for daemon set '%s' not found, so it is expected to exist already")
```
Comment (jellonek) [inline scheduler/scheduler.go]: Missing argument for '%s'. Please use `name` for it.

File: resources/daemonset.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+func (d ExistingDaemonSet) Status() (string, error) {
+	return daemonSetStatus(d.Client, d.Name)
+}
+
+func (d ExistingDaemonSet) Create() error {
+	log.Println("Looking for daemonset", d.Name)
+	status, err := d.Status()
+
+	if err == nil {
+		log.Printf("Found daemonset %s, status:%s", d.Name, status)
```
Comment (jellonek) [inline resources/daemonset.go]: Please add space after `status:`.

**Comment by pigmej** [comment]
@PiotrProkop any progres there?

**Comment by nebril** [comment]
Please add your new type to https://github.com/Mirantis/k8s-AppController/blob/master/scheduler/scheduler_test.go#L285 , it will help out with catching any changes that might have been missed.

**Comment by jellonek** [comment]
Also please add a bullet to the list in https://github.com/PiotrProkop/k8s-AppController/blob/fd5a59e3d401f2f37570a6c0915d7ae92b6ad520/README.md#resource-definitions

**Comment by nhlfr** [comment]
Please fix your rebase, I see a lot of commits from master here. Also, I'd suggest to squash your commits as well.

**Comment by jellonek** [comment]
LGTM, travis issue looks quite strange.

**Comment by PiotrProkop** [comment]
@pigmej Modify an existing example or create a new one ?

**Comment by pigmej** [comment]
I'd go with extending `extended` example

**Comment by PiotrProkop** [comment]
@pigmej done

File: resources/daemonset.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+func daemonSetKey(name string) string {
+	return "daemonset/" + name
+}
+
+func daemonSetStatus(d unversioned.DaemonSetInterface, name string) (string, error) {
+	daemonSet, err := d.Get(name)
+	if err != nil {
+		return "error", err
+	}
+	if daemonSet.Status.CurrentNumberScheduled == daemonSet.Status.DesiredNumberScheduled {
```
Comment (nebril) [inline resources/daemonset.go]: One question - are we sure we don't want to check all underlying pods status.

Comment (PiotrProkop) [inline resources/daemonset.go]: Current number scheduled returns:
```
// CurrentNumberScheduled is the number of nodes that are running at least 1
// daemon pod and are supposed to run the daemon pod.
```
but if you prefer i can check each pod separately.

Comment (nebril) [inline resources/daemonset.go]: I think it's OK to merge it as is, but we will need some additional discussion about how to implement DaemonSet status check.

**Pull request merged**